### PR TITLE
All YOLOv8 and YOLOv5 models for object detection supported

### PR DESCRIPTION
**Pull request by hdnh2006**
This PR supports all the YOLOv5 v7.0 models and all the YOLOv8 for object detection from Ultralytics.

Screenshot from 2023-04-14 11-47-20

A new class `YOLOv8` has been created inside the code [`yolov8.py`](https://github.com/hdnh2006/anylabeling/blob/93e28bd35ccfb2e1c9990d62311323c8c523c14d/anylabeling/services/auto_labeling/yolov8.py). This class will deal with the post process of YOLOv8 which is a bit different of YOLOv5.

Models must be exported with `opset=12` with the code:
```
pip install ultralytics
yolo export model=yolov8n.pt imgsz=640 format=onnx opset=12
```

However, they are available in my assets: https://github.com/hdnh2006/anylabeling-assets/releases/tag/v0.0.0.

The models must be added to @vietanhdev repo, otherwise, they need to be downloaded manually.

### CODE REVIEW

File: anylabeling/services/auto_labeling/model_manager.py [hunk 1]
```diff
@@ -107,14 +116,21 @@ def _load_model(self, model_name):
 
         if model_info["type"] == "yolov5":
             from .yolov5 import YOLOv5
-
+
```
Comment (vietanhdev) [inline anylabeling/services/auto_labeling/model_manager.py:119]: Please remove these spaces.

File: anylabeling/services/auto_labeling/model.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def get_model_abs_path(self, model_path):
                 " while..."
             )
             relative_path = model_path.replace("anylabeling_assets/", "")
-            download_url = self.BASE_DOWNLOAD_URL + relative_path
+            download_url = self.BASE_DOWNLOAD_URL + relative_path.replace("models/yolov5/","").replace("models/segment_anything/","")
```
Comment (vietanhdev) [inline anylabeling/services/auto_labeling/model.py:60]: I think replacing like this is not a best practice.
Please change the code as following:
```py
if model_path.startswith("anylabeling_assets/"):
    download_url = ....
elif model_path.lower().startswith(("http://", "https://")):
    download_url = .....
```

**Review by vietanhdev — CHANGES_REQUESTED** [review]
- Please add onnxruntime-gpu to [pyproject.toml](https://github.com/vietanhdev/anylabeling/blob/master/pyproject.toml) and [requirements-macos.txt](https://github.com/vietanhdev/anylabeling/blob/master/requirements-macos.txt) also.

**Comment by hdnh2006** [comment]
> * Please add onnxruntime-gpu to [pyproject.toml](https://github.com/vietanhdev/anylabeling/blob/master/pyproject.toml) and [requirements-macos.txt](https://github.com/vietanhdev/anylabeling/blob/master/requirements-macos.txt) also.

I added the onnnruntime-gpu to `requirements-mac.txt` but I am not too used to pyproject.toml files, so please, if you can be more specific, I would thank you.

**Review by vietanhdev — APPROVED** [review]

**Comment by vietanhdev** [comment]
@hdnh2006  "onnxruntime-gpu" is not supported on mac m1. Therefore I removed it from [pyproject.toml](https://github.com/vietanhdev/anylabeling/blob/master/pyproject.toml) and [requirements-macos.txt](https://github.com/vietanhdev/anylabeling/blob/master/requirements-macos.txt).

**Pull request merged**